File: 0852.山脉数组的峰顶索引/solution.cpp
class Solution {
public:
    int peakIndexInMountainArray(vector<int>& arr) {
        int sz=arr.size();
        int l=1,r=sz-2;
        while (l<r) {
            int mid=l+(r-l+1)/2;
            if (arr.at(mid)>arr.at(mid-1)) {
                l=mid;
            } else {
                r=mid-1;
            }
        }
        return l;
    }
};

File: 0852.山脉数组的峰顶索引/solution.go
func peakIndexInMountainArray(arr []int) int {
    sz:=len(arr)
    l,r:=1,sz-2
    for l<r {
        mid:=l+(r-l+1)/2
        if arr[mid]>arr[mid-1] {
            l=mid
        } else {
            r=mid-1
        }
    }
    return l
}